src/retrieval/inverted_index.h

```diff
@@ -329,10 +355,13 @@ void InvertedIndex<kDescType, kDescDim, kEmbeddingDim>::Read(
   ifs->read(reinterpret_cast<char*>(&num_images), sizeof(int32_t));
   CHECK_GE(num_images, 0);
 
-  normalization_constants_.resize(num_images);
-  for (int32_t image_id = 0; image_id < num_images; ++image_id) {
-    ifs->read(reinterpret_cast<char*>(&normalization_constants_[image_id]),
-              sizeof(float));
+  normalization_constants_.clear();
```

Comment:
Since the number of images is known, ``normalization_constants_`` could be preallocated.